frontend/components/idioms.jsx

```diff
@@ -52,91 +52,72 @@ export function UseEffect({ fetchURL, label }) {
 }
 
 export function UseEffectDerivedCalculation() {
-  const [remainder, setReminder] = useState()
-  const [clickedTimes, setClickedTimes] = useState()
-
-  useEffect(() => {
-    setReminder(clickedTimes % 5)
-  }, [clickedTimes])
-
-  const handleClick = () => setClickedTimes(clickedTimes + 1)
+  const [clickedTimes, setClickedTimes] = useState(0); // default state
 
+  const handleClick = () => setClickedTimes(clickedTimes + 1);
   return (
     <div>
       <button onClick={handleClick}>Add Click Count</button>
-      <span>
-        {sum}
-      </span>
-      <span>
-        {remainder}
-      </span>
+      <span>{clickedTimes}</span>
+      <span>{clickedTimes % 5}</span>
+      {/* no need for a new state for something that can be set in the line */}
     </div>
-  )
+  );
 }
 
 export function UseEffectLifeCycle() {
-  const [_loaded, setLoaded] = useState()
-
-  useEffect(() => {
-    setTimeout(() => setLoaded(true), 1000)
-  }, [])
-
-  const handleClick = () => setClickedTimes(clickedTimes + 1)
+  const [clickedTimes, setClickedTimes] = useState(0);
+  // no need of use effect here
+  const handleClick = () => setClickedTimes(clickedTimes + 1);
 
   return (
     <div>
       <button onClick={handleClick}>Add Click Count</button>
-      <span>
-        {clickedTimes}
-      </span>
+      <span>{clickedTimes}</span>
     </div>
-  )
+  );
 }
 
 export function DirtyUnmount() {
   const [time, setTime] = useState(0);
 
   useEffect(() => {
-    setInterval(() => {
-      setTime(t => t + 1)
-    }, 1000)
-  }, [])
+    const id = setInterval(() => {
+      setTime((t) => t + 1);
+    }, 1000);
+    return () => clearInterval(id); //clean it up
+  }, []);
 
-  return (
-    <div>
-      Clock in seconds: {time}
-    </div>
-  )
+  return <div>Clock in seconds: {time}</div>;
 }
 
 export function AvoidingUseState() {
+  // ref doesn't update the screen, so it if you wanna see Mounted you should use State here.
   const ref = useRef('Unmounted');
 
   useEffect(() => {
-    ref.current = "Mounted"
-  }, [])
+    ref.current = 'Mounted';
+  }, []);
 
-  return (
-    <div>
-      {ref.current}
-    </div>
-  )
+  return <div>{ref.current}</div>;
 }
 
-async function API() { return true }
+async function API() {
+  return true;
+}
 
 export function UntraceableState() {
-  const [result, setResult] = useState()  
-  let loading = false
+  const [result, setResult] = useState(''); // default
+  let loading = false;
 
   useEffect(() => {
     const fetchData = async () => {
-      loading = true
-      const result = await API();
+      loading = true;
+      const response = await API(); // let's not name it the same
       loading = false;
-      setResult(result);
+      setResult(response);
     };
```

Comment:
This change is meaninguless. Check the function name for the hint of what's wrong.